src/app/containers/CpsFeaturesAnalysis/index.jsx

```diff
@@ -80,7 +83,8 @@ PromoListComponent.defaultProps = {
 
 const PromoComponent = ({ promo, dir }) => {
   const { serviceDatetimeLocale } = useContext(ServiceContext);
-  const viewRef = useViewTracker(eventTrackingData);
+  const { optimizely } = useContext(OptimizelyContext);
```

Comment:
Could this line be moved inside the `getEventTrackingData` function? I feel like that will cause some sort of issue in react but just would be nice to not repeat the call.

Reply:
Yeah, I wanted to do that initially but React complains because hooks have to be used within React components to make sure the component lifecycle doesn't go out of whack 😢.